grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+# Lint as: python3
+from absl import app, flags
+import json
+import os
+from typing import Any, cast, Dict, List, Text, Tuple
+
+from fleetspeak.src.server.proto.fleetspeak_server import resource_pb2
+
+from grr_response_core import config
+from grr_response_core.config import server as config_server
+from grr_response_core.config import contexts
+from grr_response_server import fleetspeak_connector
+from grr_response_server import fleetspeak_utils
+from grr_response_server import server_startup
+
+from werkzeug import exceptions as werkzeug_exceptions
+from werkzeug import routing as werkzeug_routing
+from werkzeug import serving as werkzeug_serving
+from werkzeug import wrappers as werkzeug_wrappers
+from werkzeug.wrappers import json as werkzeug_wrappers_json
+
+
+AVAILABLE_METRICS = [
+    "mean_user_cpu_rate",
+    "max_user_cpu_rate",
+    "mean_system_cpu_rate",
+    "max_system_cpu_rate",
+    "mean_resident_memory_mib",
+    "max_resident_memory_mib",
+]
+
+JSON_MIME_TYPE = "application/json"
+
+flags.DEFINE_bool(
+    "version",
+    default=False,
+    allow_override=True,
+    help="Print the GRR console version number and exit immediately.")
+
+
+class JSONRequest(werkzeug_wrappers_json.JSONMixin, werkzeug_wrappers.Request):
+  pass
+
+
+class JSONResponse(werkzeug_wrappers_json.JSONMixin, werkzeug_wrappers.Response):
```

Comment:
`JSONMixin` actually does not affect constructing a Response, see [json.py](https://github.com/pallets/werkzeug/blob/master/src/werkzeug/wrappers/json.py). The JSONMixin is only used for parsing JSON. This is reflected in your following code, where you need to call `json.dumps` and setting the mimetype manually every time.

I suggest removing `JSONMixin` from `JSONResponse` and overriding `__init__` to call `json.dumps` on whatever data is passed as `response` parameter. Also, the new constructor should always set the mimetype.

Reply:
I agree that `JSONMixin` does not affect the response, but its property `json` is utilized later within the tests for both endpoints `/search` and `/query`. Also, not using `json` will cause us to decode the JSON bytes ourselves from the response.
Therefore, I think it is wise to leave the inheritance as is and get this property from the mixin.

Re overriding `__init__` and modifying the relevant parameters: I agree. PTAL on this and let me know what you think.